xDrip/Scenes/Entries/EntriesList/EntriesListPresenter.swift

```diff
@@ -14,15 +14,74 @@ import UIKit
 
 protocol EntriesListPresentationLogic {
     func presentLoad(response: EntriesList.Load.Response)
+    func inject(formattingWorker: EntriesListFormattingWorker)
 }
 
 final class EntriesListPresenter: EntriesListPresentationLogic {
     weak var viewController: EntriesListDisplayLogic?
+    private var formattingWorker: EntriesListFormattingWorker?
     
     // MARK: Do something
     
     func presentLoad(response: EntriesList.Load.Response) {
-        let viewModel = EntriesList.Load.ViewModel()
+        let entries = response.entries
+        let cellViewModel = formattingWorker?.formatEntries(entries) ?? []
+        
+        let title = "entries_list_data_section_title".localized.uppercased()
+        
+        let viewModel = EntriesList.Load.ViewModel(items: [EntriesList.SectionViewModel(title: title, items: cellViewModel)])
         viewController?.displayLoad(viewModel: viewModel)
     }
+    
+    func inject(formattingWorker: EntriesListFormattingWorker) {
+        self.formattingWorker = formattingWorker
+    }
+}
+
+protocol EntriesListFormattingWorker {
+    func formatEntries(_ entries: [AbstractEntry]) -> [EntriesList.CellViewModel]
+}
+
+final class EntriesListCarbsFormattingWorker: EntriesListFormattingWorker {
+    
+    func formatEntries(_ entries: [AbstractEntry]) -> [EntriesList.CellViewModel] {
+        var cellViewModels: [EntriesList.CellViewModel] = []
+        
+        entries.forEach { (entry) in
+            var value = ""
+
+            value = String(format: "%.02f g", (entry as! CarbEntry).amount)
+
+            let date = DateFormatter.localizedString(from: entry.date ?? Date(), dateStyle: .short, timeStyle: .short)
+
+            cellViewModels.append(
+                EntriesList.CellViewModel(value: value,
```

Comment:
To format entries correctly, you should use specific fields of `CarbsEntry` in this example. Do `entries.compactMap { $0 as? CarbEntry }` first.